src/linux/disk.rs

```diff
@@ -26,36 +28,56 @@ pub enum DiskType {
 impl From<isize> for DiskType {
     fn from(t: isize) -> DiskType {
         match t {
-            0 => DiskType::HDD,
-            1 => DiskType::SSD,
+            0 => DiskType::SSD,
```

Comment:
In testing the logic seemed backwards; if we look at `/sys/block/sda/queue/rotational` and it contains a 1, that means the device is spinning rust, i.e. an HDD.